### PR TITLE
Update involucro to be built from source

### PR DESCRIPTION
**Pull request by mbargull**
Our `involucro` package currently just repackages pre-built binaries, but we could/should just build from source instead.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/involucro/build.sh [hunk 1]
```diff
@@ -1,5 +1,52 @@
...
 
-mkdir -p $PREFIX/bin
-cp ./involucr* $PREFIX/bin/involucro
-chmod +x $PREFIX/bin/involucro
+export GOPATH="$( pwd )"
+
+module_path="$( go env GOPATH )"/src/github.com/involucro/involucro/cmd/involucro
+GOBIN="$( go env GOBIN )" \
+  go get \
+  -ldflags="-s -X github.com/involucro/involucro/app.version=${PKG_VERSION}" \
```
Comment (mbargull) [inline recipes/involucro/build.sh:8]: Upstream source has a placeholder `var version = "vDev"` which we fill out here.

File: recipes/involucro/build.sh [hunk 2]
```diff
@@ -1,5 +1,52 @@
...
+--------------------------------------------------------------------------------
+${2#${1%/}/}
+================================================================================
+EOF
+cat "${2}"
+' -- "${acc_dir}" >> "${output}"
+  rm -r "${acc_dir}" "${tmp_dir}"
+}
+
+gather_licenses ./licenses.txt "${module_path}"
```
Comment (mbargull) [inline recipes/involucro/build.sh:52]: Yeah, I know this is a little bonkers.. I've used this function in some Go packages I had to put into conda-forge recently. I'll clean it up when we have better support for this in `conda-build`.

File: recipes/involucro/meta.yaml [hunk 3]
```diff
@@ -3,24 +3,37 @@ package:
...
+  - url: https://github.com/involucro/involucro/archive/v1.1.2.tar.gz
+    sha256: be8d1b65e4841bc1ae68e418aef00cbc19052038415858f4cceb44d100dbefc0
+    folder: src/github.com/involucro/involucro
+  # Add vendor-vendor sources for missing license files.
+  - url: https://github.com/mattn/go-isatty/archive/v0.0.1.tar.gz
+    sha256: 71e6931ff2804ef711eae1882fe95beee5b60c38d929c638d553ba143a3bc7ec
+    folder: src/github.com/involucro/involucro/vendor/github.com/fatih/color/vendor/github.com/mattn/go-isatty
+  - url: https://github.com/mattn/go-colorable/archive/v0.0.1.tar.gz
+    sha256: 626e0dcec24fd3621e40eceb33791747f0970c248500930a1461f599acd1aadd
+    folder: src/github.com/involucro/involucro/vendor/github.com/fatih/color/vendor/github.com/mattn/go-colorable
```
Comment (mbargull) [inline recipes/involucro/meta.yaml:15]: This is purely for the `LICENSE` files and can be removed when/if involucro updates its dependencies upstream.

File: recipes/involucro/meta.yaml [hunk 4]
```diff
@@ -3,24 +3,37 @@ package:
...
 
 build:
-  number: 1
+  number: 2
+
+requirements:
+  build:
+    - {{ compiler('go') }}
+    - go-licenses
```
Comment (mbargull) [inline recipes/involucro/meta.yaml:23]: Interestingly enough, I got a segfault when I compiled with `{{ compiler('cgo') }}`. But we can have this as a fully statically linked binary anyway, so no need to worry there.

**Comment by mbargull** [comment]
@BiocondaBot, please fetch artifacts

**Review by bgruening — APPROVED** [review]

**Review by tomkinsc — APPROVED** [review]

**Comment by tomkinsc** [comment]
Amazing; thanks for doing all of this!

**Comment by tomkinsc** [comment]
@bioconda-bot please merge

**Pull request merged**